test/functional/test_runner.py

```diff
@@ -325,6 +326,13 @@ def run_tests(test_list, src_dir, build_dir, exeext, tmpdir, jobs=1, enable_cove
             print("\n%s%s%s failed, Duration: %s s\n" % (BOLD[1], test_result.name, BOLD[0], test_result.time))
             print(BOLD[1] + 'stdout:\n' + BOLD[0] + stdout + '\n')
             print(BOLD[1] + 'stderr:\n' + BOLD[0] + stderr + '\n')
+            if combined_logs_len and os.path.isdir(testdir):
+                # Print the final `combinedlogslen` lines of the combined logs
+                print('{}Combine the logs and print the last {} lines ...{}'.format(BOLD[1], combined_logs_len, BOLD[0]))
+                print('\n============')
+                print('{}Combined log for {}:{}'.format(BOLD[1], testdir, BOLD[0]))
+                print('============\n')
+                subprocess.run(["{} -c {} | tail -{}".format(tests_dir + './combine_logs.py', testdir, combined_logs_len)], shell=True, universal_newlines=True)
```

Comment:
Sorry I missed this when looking at that branch initially, but I doubt this will work on Windows.

Since there are plans to work on native Windows builds (https://github.com/bitcoin/bitcoin/pull/11526#issuecomment-345606323) and hopefully tests, I'd prefer if we avoided the code churn.

Reply:
Also, subprocess.run was introduced in python 3.5. However, our supported version is 3.4

Reply:
yeah - I noticed that too :(

Pushed a new version which I think should fix both problems:
- use `subprocess.Popen().communicate()`
- don't use the braindead `shell=True` argument and actually pass in arguments to the subprocess properly